App.js

```diff
@@ -0,0 +1,88 @@
+import React, {Component} from 'react';
+import {AppRegistry, Image} from 'react-native';
+import {createAppContainer, createSwitchNavigator} from 'react-navigation';
+import {createStackNavigator} from 'react-navigation-stack';
+import {createBottomTabNavigator} from 'react-navigation-tabs';
+import SplashScreen from 'react-native-splash-screen';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+// Components
+import HomeView from './components/HomeView/HomeView';
+import AccountView from './components/AccountView/AccountView';
+import StudyView from './components/StudyView/StudyView';
+import SettingsView from './components/SettingsView/SettingsView';
+import AuthView from './components/AuthView/AuthView';
+
+const HomeNavigator = createStackNavigator({
+  HomeView: {
+    screen: HomeView,
+  },
+  StudyView: {
+    // When you click a study: onPress={() => this.props.navigation.navigate('ConnectionView' .....
+    screen: StudyView,
+  },
+});
+
+const AccountNavigator = createStackNavigator({
+  Account: {
+    screen: AccountView,
+  },
+  Study: {
+    // When you click a study: onPress={() => navigate('StudyView' .....
+    screen: StudyView,
+  },
+});
```

Comment:
Just a general q about the stack handling here — the most common setup for stacks is that you have one stack per nav item, and when you're inside this stack, you can continue to push new views (of any type) on top of the stack.

In Spotify, for example, this is what allows you to go to "Home", tap a playlist, then go to "Search", search for something, then go back to home and see that playlist you were looking at. The "Home" stack retains memory of everything that's been pushed into it, and going "back" in a stack removes the most recent view. Meanwhile, the "Search" stack maintains its own state as well.

Kinda hard to describe, I suppose. But I _think_ what you'd probably want is to remove the idea of "Study" as its own stack, and `navigate.push` "Study" onto existing stacks. This way, a user can search for a study, switch to their account, look at a _different_ study there, then switch back to searching without losing their place.

Reply:
I thought that that is what I was doing. Because both the Home and Study exist under the HomeNavigator stack, wouldn't it work the way you are describing?

Reply:
Hmm, I may be thinking about this the wrong way. Looking at the docs, my take was:

- Each screen in a bottom tab navigator is typically a "stack"
- You don't need to predefine all of the items that can possibly be in a stack, in most cases you just need to define one screen and you `navigate.push` any other screens onto it

But I could be quite wrong — might be thinking about how it works in another lib, or misreading this. @solomonhawk you've worked with RN navigation more recently than I have, any thoughts?